contrib/pkg/clusterpool/clusterclaim.go

```diff
@@ -0,0 +1,106 @@
+package clusterpool
+
+import (
+	"fmt"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+
+	"github.com/openshift/hive/contrib/pkg/utils"
+	"github.com/openshift/hive/pkg/apis"
+	hivev1 "github.com/openshift/hive/pkg/apis/hive/v1"
+)
+
+type ClusterClaimOptions struct {
+	Namespace       string
+	Lifetime        time.Duration
+	ClusterPoolName string
+
+	homeDir string
+}
+
+func NewClaimClusterPoolCommand() *cobra.Command {
+	opt := &ClusterClaimOptions{}
+
+	opt.homeDir = "."
+
+	defaultDuration, _ := time.ParseDuration("8h")
+
+	cmd := &cobra.Command{
+		Use:   "claim CLUSTER_POOL_NAME",
+		Short: "claim a cluster from a clusterpool",
+		Long:  "claims a cluster from the clusterpool in the given namespace",
+		Args:  cobra.ExactArgs(1),
+		Run: func(cmd *cobra.Command, args []string) {
+			log.SetLevel(log.InfoLevel)
+			opt.ClusterPoolName = args[0]
+			err := opt.run()
+			if err != nil {
+				log.WithError(err).Fatal("Error")
+			}
+		},
+	}
+
+	flags := cmd.Flags()
+	flags.StringVarP(&opt.Namespace, "namespace", "n", "",
+		"Namespace to create cluster claim in. Has to be the namespace in which the cluster pool is deployed")
+	flags.DurationVar(&opt.Lifetime, "lifetime", defaultDuration, "Lifetime of the cluster claim. "+
+		"Default is 8 hours")
+
+	return cmd
+}
+
+func (o ClusterClaimOptions) run() error {
+	if err := apis.AddToScheme(scheme.Scheme); err != nil {
+		return err
+	}
+	claim := o.generateClaim()
+
+	rh, err := utils.GetResourceHelper("claim")
+	if err != nil {
+		return err
+	}
+	if len(o.Namespace) == 0 {
+		o.Namespace, err = utils.DefaultNamespace()
+		if err != nil {
+			log.Error("Cannot determine default namespace")
+			return err
+		}
+	}
+	accessor, err := meta.Accessor(claim)
+	if err != nil {
+		log.WithError(err).Errorf("Cannot create accessor for object of type %T", claim)
+		return err
+	}
+	accessor.SetNamespace(o.Namespace)
+	if _, err := rh.ApplyRuntimeObject(claim, scheme.Scheme); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (o ClusterClaimOptions) generateClaim() *hivev1.ClusterClaim {
+	return &hivev1.ClusterClaim{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ClusterClaim",
+			APIVersion: hivev1.SchemeGroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        o.getClusterClaimName(),
+			Annotations: map[string]string{},
+		},
+		Spec: hivev1.ClusterClaimSpec{
+			ClusterPoolName: o.ClusterPoolName,
+			Lifetime:        &metav1.Duration{Duration: o.Lifetime},
+		},
+	}
+}
+
+func (o ClusterClaimOptions) getClusterClaimName() string {
+	return fmt.Sprintf("%s-claim", o.ClusterPoolName)
```

Comment:
The user needs to be able to specify the name that they want for the ClusterClaim. Otherwise, the user will only ever be able to create one per ClusterPool, as the name must be unique.